src/lib/config.ts

```diff
@@ -1269,11 +1271,38 @@ export function unset(...target) {
 export async function save(storage: "local" | "sync" = get("storageloc")) {
     // let storageobj = storage === "local" ? browser.storage.local : browser.storage.sync
     // storageobj.set({CONFIGNAME: USERCONFIG})
+    await locks.acquire("config")
     const settingsobj = o({})
     settingsobj[CONFIGNAME] = USERCONFIG
-    return storage === "local"
+    const ans = storage === "local"
         ? browser.storage.local.set(settingsobj)
         : browser.storage.sync.set(settingsobj)
+
+    // Ignore errors about background pages that aren't listening
```

Comment:
Isn't there a single background page and isn't it always listening?

Reply:
I don't know if it's possible to have more than one background page running - I think we've had bugs in the past that suggested that that sometimes happened.

It only listens once `browser.runtime.onMessage.addListener` is called, which isn't always true when Tridactyl is starting up.